src/test/common/pluginTester.ts

```diff
@@ -0,0 +1,11 @@
+import { IMessage } from '../../common/types';
+
+export abstract class PluginTester {
+  public abstract channelName: string;
+  public abstract args: string;
+
+  constructor() {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public onResponse(message: IMessage): void {}
```

Comment:
I think this should be marked as `abstract` because it seems like a requirement. I believe this would also prevent you from having to suppress the unused-var error.

Reply:
its not required, lots of plugins wouldn't need something done to the result such as a react or something. Most of the time i was using this, i didnt need the `onReponse`

Reply:
```typescript
if (!testCase.onResponse) {
      return;
} 
```

I see this code, but shouldn't this never be false since `onResponse` is always inherited?

Reply:
oh yeah, i think its left over. I had tried
```ts
public abstract onReponse? (...)
```
but didn't get that to work

Reply:
Now that I think of it PluginTester is better off just as an interface. Since the base class doesn't contain any business logic. You can set your `onResponse` type to `onResponse?: (message: IMessage) => void`

Reply:
That's a good idea. Thank you :)

Reply:
Ok cool, correct me if I'm wrong but the ts error suppression comment isn't needed anymore, right?